app/scaffold-controllers/error-screen/errorController.js

```diff
@@ -0,0 +1,112 @@
+define([
+    'astro-full',
+    'astro-rpc',
+    'bluebird',
+    'config/errorConfig',
+    'config/baseConfig',
+    'plugins/webViewPlugin',
+    'plugins/modalViewPlugin'
+/* eslint-disable */
+], function(
+    Astro,
+    AstroRpc,
+    Promise,
+    ErrorConfig,
+    BaseConfig,
+    WebViewPlugin,
+    ModalViewPlugin
+) {
+/* eslint-enable */
+
+    var ErrorController = function(modalView, webView) {
+        this.isShowing = false;
+        this.viewPlugin = webView;
+        this.modalView = modalView;
+        this.errorType = null;
+    };
+
+    ErrorController.init = function() {
+        return Promise.join(
+            WebViewPlugin.init(),
+            ModalViewPlugin.init(),
+        function(webView, modalView) {
+            var loader = webView.getLoader();
+            loader.setColor(BaseConfig.loaderColor);
+            webView.disableScrollBounce();
+            modalView.setContentView(webView);
+
+            var errorController = new ErrorController(modalView, webView);
+            Astro.registerRpcMethod(AstroRpc.names.errorContent, [], function(res) {
+                res.send(null, errorController.errorContent());
+            });
+            return errorController;
+        });
+    };
+
+    ErrorController.prototype.show = function() {
+        if (this.isShowing) {
+            return;
+        }
+        this.isShowing = true;
+        this.modalView.show({animated: true});
+    };
+
+    ErrorController.prototype.hide = function() {
+        this.isShowing = false;
+        this.modalView.hide({animated: true});
+    };
+
+    ErrorController.prototype.errorContent = function() {
+        if (!this.errorType) {
+            return null;
+        }
+        return ErrorConfig.errors[this.errorType];
+    };
+
+    // Remove the events once the modal is hidden or else some untriggered
+    // once events will still exist on the errorModal page and create
```

Comment:
Nitpick... can we quote "once" ... I had to read this 3 times to understand we were talking about `once()` events. :)